sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/Resource.Serialization.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing the base resource used by all azure resources.
+    /// </summary>
+    public abstract partial class Resource
+    {
+        /// <summary>
+        /// Serialize the input Resource object.
+        /// </summary>
+        /// <param name="writer"> Input Json writer. </param>
+        /// <param name="value"> Input Resource object. </param>
+        internal static void Serialize(Utf8JsonWriter writer, Resource value)
```

Comment:
let's follow the private write() and internal static DeserializeXXX() pattern. Can see existing RP code.

Reply:
Done